src/aio.rs

```diff
@@ -577,3 +577,91 @@ impl ConnectionLike for MultiplexedConnection {
         self.db
     }
 }
+
+#[cfg(feature = "tokio-rt-core")]
+mod connection_manager {
+    use super::*;
+
+    /// A `ConnectionManager` is a proxy that wraps a multiplexed connection and
+    /// automatically reconnects to the server when necessary.
+    pub struct ConnectionManager {
+        connection_info: ConnectionInfo,
+        connection: MultiplexedConnection,
+    }
+
+    #[cfg(feature = "tokio-rt-core")]
+    impl ConnectionManager {
+        /// Connect to the server and store the connection inside the returned `ConnectionManager`.
+        ///
+        /// TODO: Add `new_tokio`?
```

Comment:
I'm not sure if this could easily be implemented non-tokio-specific, the futures from the old and the new connection would need to be proxied somehow. If that's possible, we could split `new` and `new_tokio`.

Reply:
Just need to propagate `driver` out to the caller again for this.

Reply:
But the `driver` is replaced when reconnecting.

Reply:
Aha, that's true. Could perhaps use https://docs.rs/futures/0.3.1/futures/task/trait.Spawn.html but tokio unfortunately does not implement it.

Reply:
Ok, then I guess we have to require Tokio for now.